Update changelog server.md for 24.1 EN (#10852)
ydb-platform/ydb: ydb/docs/en/core/changelog-server.md

# {{ ydb-short-name }} Server changelog

## Version 24.1 {#24-1}

Release date: July 31, 2024.

> ### Functionality
> 
> * The [Knn UDF](./yql/reference/udf/list/knn.md) function for precise nearest vector search has been implemented.
> * The QueryService gRPC service has been developed, providing the ability to execute all types of queries (DML, DDL) and to retrieve unlimited amounts of data.
> * [Integration with the LDAP protocol](./concepts/auth.md) has been implemented, and it is now possible to obtain a list of groups from external LDAP directories.

Reviewer: ```suggestion
### Functionality

* The [Knn UDF](./yql/reference/udf/list/knn.md) function for precise nearest vector search has been implemented.
* The gRPC Query service has been developed, enabling the execution of all types of queries (DML, DDL) and retrieval of unlimited amounts of data.
* [Integration with the LDAP protocol](./concepts/auth.md) has been implemented, allowing the retrieval of a list of groups from external LDAP directories.
```

> ### Embedded UI
> 
> * A resource consumption diagnostic dashboard has been added, which is located on the database information tab and allows you to determine the current state of consumption of the main resources: processor cores, RAM, and space in the network distributed storage.
> * Charts for monitoring the key performance indicators of the {{ ydb-short-name }} cluster have been added.

Reviewer: ```suggestion
### Embedded UI

* The database information tab now includes a resource consumption diagnostic dashboard, which allows users to assess the current consumption of key resources: processor cores, RAM, and distributed storage space.
* Charts for monitoring the key performance indicators of the {{ ydb-short-name }} cluster have been added.
```

> ### Performance
> 
> * [Sessions timeouts](https://github.com/ydb-platform/ydb/pull/1837) for the coordination service from the server to the client have been optimized. Previously, the timeout was 5 seconds, which in the worst case led to identifying an unresponsive client (and releasing the resources it was holding) within 10 seconds. In the new version, the check time depends on the session waiting time, which provides a faster response when changing the leader or acquiring distributed locks.
> * CPU consumption by [SchemeShard](./concepts/glossary.md#scheme-shard) replicas has been [optimized](https://github.com/ydb-platform/ydb/pull/2391), especially when processing fast updates for tables with a large number of partitions.

Reviewer: ```suggestion
### Performance

* [Session timeouts](https://github.com/ydb-platform/ydb/pull/1837) for the coordination service between server and client have been optimized. Previously, the timeout was 5 seconds, which could result in a 10-second delay in identifying an unresponsive client and releasing its resources. In the new version, the check interval depends on the session's wait time, allowing for faster responses during leader changes or when acquiring distributed locks.
* CPU consumption by [SchemeShard](./concepts/glossary.md#scheme-shard) replicas has been [optimized](https://github.com/ydb-platform/ydb/pull/2391), particularly when handling rapid updates for tables with a large number of partitions.
```

> ### Bug fixes
> 
> * The error of a possible queue overflow has been [fixed](https://github.com/ydb-platform/ydb/pull/3917). [Change Data Capture](./dev/cdc.md) reserves the capacity of the change queue during the initial scan.
> * A potential deadlock between receiving CDC records and sending them has been [fixed](https://github.com/ydb-platform/ydb/pull/4597).
> * The issue of losing the mediator task queue when reconnecting the mediator has been [fixed](https://github.com/ydb-platform/ydb/pull/2056). The fix allows processing the mediator task queue during resynchronization.
> * A rarely occurring error has been [fixed](https://github.com/ydb-platform/ydb/pull/2624), when with volatile transactions enabled and used, a successful transaction confirmation result was returned before it was successfully committed. Volatile transactions are turned off by default and are under development.
> * A rare error has been [fixed](https://github.com/ydb-platform/ydb/pull/2839) that led to the loss of established locks and the successful confirmation of transactions that should have failed with the Transaction Locks Invalidated error.
> * A rare error that could lead to a possible violation of data integrity guarantees during concurrent writing and reading data by a specific key has been [fixed](https://github.com/ydb-platform/ydb/pull/3074).
> * The problem has been [fixed](https://github.com/ydb-platform/ydb/pull/4343), due to which read replicas stopped processing requests.
> * A rare error has been [fixed](https://github.com/ydb-platform/ydb/pull/4979) that could lead to database processes terminating abnormally if there were uncommitted transactions on a table at the time of its renaming.
> * An error in the logic of determining the status of a static group has been [fixed](https://github.com/ydb-platform/ydb/pull/3632), when a static group was not marked as non-working, although it should have been.
> * An error of partial commit of a distributed transaction with uncommitted changes in case of some races with restarts has been [fixed](https://github.com/ydb-platform/ydb/pull/2169).
> * Anomalies with reading outdated data, which were [detected using Jepsen](https://blog.ydb.tech/hardening-ydb-with-jepsen-lessons-learned-e3238a7ef4f2), have been [fixed](https://github.com/ydb-platform/ydb/pull/2374).

Reviewer: ```suggestion
### Bug fixes

* A possible queue overflow error has been [fixed](https://github.com/ydb-platform/ydb/pull/3917). [Change Data Capture](./dev/cdc.md) now reserves the change queue capacity during the initial scan.
* A potential deadlock between receiving and sending CDC records has been [fixed](https://github.com/ydb-platform/ydb/pull/4597).
* An issue causing the loss of the mediator task queue during mediator reconnection has been [fixed](https://github.com/ydb-platform/ydb/pull/2056). This fix allows processing of the mediator task queue during resynchronization.
* A rarely occurring error has been [fixed](https://github.com/ydb-platform/ydb/pull/2624), where with volatile transactions enabled, a successful transaction confirmation result could be returned before the transaction was fully committed. Volatile transactions remain disabled by default and are still under development.
* A rare error that led to the loss of established locks and the successful confirmation of transactions that should have failed with a "Transaction Locks Invalidated" error has been [fixed](https://github.com/ydb-platform/ydb/pull/2839).
* A rare error that could result in a violation of data integrity guarantees during concurrent read and write operations on a specific key has been [fixed](https://github.com/ydb-platform/ydb/pull/3074).
* An issue causing read replicas to stop processing requests has been [fixed](https://github.com/ydb-platform/ydb/pull/4343).
* A rare error that could cause abnormal termination of database processes if there were uncommitted transactions on a table during its renaming has been [fixed](https://github.com/ydb-platform/ydb/pull/4979).
* An error in determining the status of a static group, where it was not marked as non-working when it should have been, has been [fixed](https://github.com/ydb-platform/ydb/pull/3632).
* An error involving partial commits of a distributed transaction with uncommitted changes, caused by certain race conditions with restarts, has been [fixed](https://github.com/ydb-platform/ydb/pull/2169).
* Anomalies related to reading outdated data, [detected using Jepsen](https://blog.ydb.tech/hardening-ydb-with-jepsen-lessons-learned-e3238a7ef4f2), have been [fixed](https://github.com/ydb-platform/ydb/pull/2374).
```
